lib/rubocop/cop/layout/closing_heredoc_indentation.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Layout
+      #
+      # Checks the indentation of here document closings.
+      #
+      # @example
+      #   # bad
+      #
+      #   class Foo
+      #     def bar
+      #       <<~SQL
+      #         'Hi'
+      #     SQL
+      #     end
+      #   end
+      #
+      #   # good
+      #
+      #   class Foo
+      #     def bar
+      #       <<~SQL
+      #         'Hi'
+      #       SQL
+      #     end
+      #   end
+      #
+      class ClosingHeredocIndentation < Cop
+        include Heredoc
+
+        MSG = '`%<closing>s` is not aligned with `%<opening>s`.'.freeze
+
+        def on_heredoc(node)
+          return if opening_indentation(node) == closing_indentation(node)
+
+          msg = format(
+            MSG,
+            closing: heredoc_closing(node).strip,
+            opening: heredoc_opening(node).strip
+          )
+          add_offense(node, location: :heredoc_end, message: msg)
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            corrector.replace(node.loc.heredoc_end, indented_end(node))
+          end
+        end
+
+        private
+
+        def indented_end(node)
+          closing_indent = closing_indentation(node)
+          opening_indent = opening_indentation(node)
+          closing_text = heredoc_closing(node)
+          closing_text.gsub(/^\s{#{closing_indent}}/, ' ' * opening_indent)
+        end
+
+        def opening_indentation(node)
+          indent_level(heredoc_opening(node))
+        end
+
+        def closing_indentation(node)
+          indent_level(heredoc_closing(node))
+        end
+
+        def heredoc_opening(node)
+          node.loc.expression.source_line
+        end
+
+        def heredoc_closing(node)
+          node.loc.heredoc_end.source_line
+        end
+
+        def indent_level(str)
+          str[/\A */].size
```

Comment:
I tend to prefer using `#length` for strings, as it makes more sense, conceptually. 🙂